src/core/modules/Planet.cpp

```diff
@@ -4890,10 +4893,40 @@ void Planet::drawOrbit(const StelCore* core)
 
 	sPainter.setColor(getCurrentOrbitColor(), orbitFader.getInterstate());
 	Vec3d onscreen;
+
+	if (!fromMoonPerspective) {
+		// Update the orbit positions to the current planet date.
+		computeOrbit();
+	}
+	else {
+		double dateJDE = lastJDE;
+		double calc_date;
+
+		Vec3d myPos = core->getCurrentPlanet()->getHeliocentricEclipticPos(lastJDE);
+		Vec3d myparentPos = core->getCurrentPlanet()->getParent()->getHeliocentricEclipticPos(dateJDE);
+
+		// pretend they are on x-y plane
+		myPos[2] = 0;
+		myparentPos[2] = 0;
+
+		// angle between me and parent
+		double theta = atan2(myPos[1], myPos[0]) - atan2(myparentPos[1], myparentPos[0]);
+		theta = theta / M_PI * 180.;  // should concentrate on this theta when sampling orbit
+
+		double f;
+		for(int d = 0; d < ORBIT_SEGMENTS; d++)
+		{
+			f = static_cast<double>(d - ORBIT_SEGMENTS/2.) / (ORBIT_SEGMENTS/2.);
+			// make sure only sample half of the orbit forward and half backward
+			// the sampling spacing is trial and error, but power of 13 seems to be good (i.e., densely sample around the current date)
+			calc_date = dateJDE + pow(f, 13)*deltaOrbitJDE*ORBIT_SEGMENTS/2. + theta/180.*deltaOrbitJDE*ORBIT_SEGMENTS/2;
+			orbit[d] = getEclipticPos(calc_date);
+		}
+	}
 	const Vec3d savePos = orbit[ORBIT_SEGMENTS/2];
 	if (closeOrbit)
 	{
-		if (!keplerOrbit || keplerOrbit->getEccentricity()<=0.3)
+		if ((!keplerOrbit || keplerOrbit->getEccentricity()<=0.3) && !fromMoonPerspective)
```

Comment:
Wait! Please add braces to avoid possible problems

Reply:
Fixed